From 4f83a642dced199a97b2af7e136d3be4addc77e7 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Thu, 9 Aug 2018 23:08:53 +0200
Subject: [PATCH 09/11] soong: Squash of lineage-sdk bringup commits

Squash of:

Author: Rashed Abdel-Tawab <rashed@linux.com>
Date:   Thu Aug 9 14:08:53 2018 -0700
    soong: Special case Lineage SDK
     * org.lineageos.platform-res.apk needs to be installed to /system/framework
     * org.lineageos.platform-res needs to be a dependency for
       org.lineageos.platform and org.lineageos.platform.internal
     * Add other special exceptions for org.lineageos.platform-res
    Change-Id: Ic617c07c086916005ea4b88f26d31c61691a45f8

Author: Sam Mortimer <sam@mortimer.me.uk>
Date:   Thu Aug 30 15:33:16 2018 -0700
    soong: make org.lineage.platform-res depend on framework-res
    *) Allows us to build org.lineage.platform-res with no_framework_libs
       true (as is done for framework-res).
    *) Whilst we're here, undo a dependency loop prevention in aar.go
       that we added during sdk bringup to allow our platform res to build
       with no_framework_libs false.
    Change-Id: Ib452a2e45112baf5d61b70b4be1ce0c01dfd84e5

Author: Luca Stefani <luca.stefani.ge1@gmail.com>
Date:   Mon Feb 4 18:56:52 2019 +0100
    Always link org.lineageos.platform-res for org.lineageos.platform.sdk
    Test: m clean && m org.lineageos.platform.sdk
    Change-Id: I58956855bd4d1157e2582103c4861e7b384b4f73

Author: Sam Mortimer <sam@mortimer.me.uk>
Date:   Fri Aug 31 10:52:29 2018 -0700
    soong: Allow framework to access lineage-sdk resources
    *) Make framework depend on lineage-sdk resource package
    *) Allows framework module to access lineage-sdk resources
       via usual org.lineageos.platform.internal.R paths.
    Change-Id: Ifd19d43d9308ac370ad40a499de16bf8ce204beb

Change-Id: Icc18de5dfaa83fc0a1eda6f3704f3a92e1de0764
---
 java/base.go | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/java/base.go b/java/base.go
index df70efb44..39f5639bf 100644
--- a/java/base.go
+++ b/java/base.go
@@ -1648,6 +1648,20 @@ func (j *Module) collectDeps(ctx android.ModuleContext) deps {
 			return
 		}
 
+		if tag == lineageResTag {
+			if ctx.ModuleName() == "org.lineageos.platform" ||
+				ctx.ModuleName() == "org.lineageos.platform.internal" ||
+				ctx.ModuleName() == "org.lineageos.platform.sdk" {
+				// org.lineageos.platform.jar has a one-off dependency on the R.java and Manifest.java files
+				// generated by org.lineageos.platform-res.apk
+				deps.srcJars = append(deps.srcJars, module.(*AndroidApp).aaptSrcJar)
+			}
+			if ctx.ModuleName() == "framework" {
+				// framework.jar has a one-off dependency on the R.java and Manifest.java files
+				// generated by org.lineageos.platform-res.apk
+				deps.srcJars = append(deps.srcJars, module.(*AndroidApp).aaptSrcJar)
+			}
+		}
 		if dep, ok := module.(SdkLibraryDependency); ok {
 			switch tag {
 			case libTag:
-- 
2.37.1

